extension/agenthealth/handler/stats/provider/statuscode.go

```diff
@@ -0,0 +1,180 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package provider
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/amazon-contributing/opentelemetry-collector-contrib/extension/awsmiddleware"
+
+	"github.com/aws/amazon-cloudwatch-agent/extension/agenthealth/handler/stats/agent"
+)
+
+const (
+	statusResetInterval = 5 * time.Minute
+	statusHandlerID     = "cloudwatchagent.StatusCodeHandler"
+)
+
+var (
+	statusCodeSingleton *StatusCodeHandler
+	statusCodeStatsOnce sync.Once
+)
+
+// StatusCodeHandler provides monitoring for status codes per operation.
+type StatusCodeHandler struct {
+	statsByOperation sync.Map
```

Comment:
It looks like you're locking the mutex anytime you access the map so it probably doesn't need to be a sync map

Reply:
removed mutex since we have sync.map